### PR TITLE
[extractor/youtube] Download manifestless post-live videos from the start

### PR DESCRIPTION
**Pull request by Lesmiscore**
**IMPORTANT**: PRs without the template will be CLOSED

### Description of your *pull request* and other information

</details>

This allows you to download whole archived livestream (longer than 4hrs), even though it's immediately after it finishes.

Known problems:
- [x] ~~It currently shows `Total fragment: unknown (live)`.~~
	- Since number of total fragments won't increase, maybe it may force enumerate all the fragments first

<details open><summary>Template</summary> 

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [contributing guidelines](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions) including [yt-dlp coding conventions](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#yt-dlp-coding-conventions)
- [x] [Searched](https://github.com/yt-dlp/yt-dlp/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8) and [ran relevant tests](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#developer-instructions)

### In order to be accepted and merged into yt-dlp each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Fix or improvement to an extractor (Make sure to add/update tests)
- [ ] New extractor ([Piracy websites will not be accepted](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#is-the-website-primarily-used-for-piracy))
- [ ] Core bug fix/improvement
- [ ] New feature (It is strongly [recommended to open an issue first](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#adding-new-feature-or-making-overarching-changes))

### CODE REVIEW

**Comment by pukkandan** [comment]
I've made the fragments a LazyList, so now we get the best of both generator and list. Pls go through my changes. If everything is okay with you, this can be merged

**Review by pukkandan — APPROVED** [review]

File: yt_dlp/extractor/youtube.py [hunk 1]
```diff
@@ -3652,11 +3679,23 @@ def _list_formats(self, video_id, microformats, video_details, player_responses,
...
+        if is_live is None and is_upcoming or live_content is False:
+            is_live = False
+        if is_upcoming is None and (live_content or is_live):
+            is_upcoming = False
+        post_live = get_first(video_details, 'isPostLiveDvr')
+        live_status = ('post_live' if post_live
+                       else 'is_live' if is_live
+                       else 'is_upcoming' if is_upcoming
+                       else None if None in (is_live, is_upcoming, live_content)
+                       else 'was_live' if live_content else 'not_live')
```
Comment (pukkandan) [inline yt_dlp/extractor/youtube.py:3693]: would appreciate if someone can double check this logic

File: yt_dlp/downloader/__init__.py [hunk 2]
```diff
@@ -13,7 +13,8 @@ def get_suitable_downloader(info_dict, params={}, default=NO_DEFAULT, protocol=N
         return FFmpegFD
     elif (set(downloaders) == {DashSegmentsFD}
           and not (to_stdout and len(protocols) > 1)
-          and set(protocols) == {'http_dash_segments_generator'}):
+          and set(protocols) == {'http_dash_segments_generator'}
+          and info_dict.get('is_live')):
```
[the remark this answers is not in the document]
Comment (Lesmiscore) [reply yt_dlp/downloader/__init__.py]: Why? Reverting there will run simultaneous download which is unnecessary for this

[the remark this answers is not in the document]
Comment (pukkandan) [reply yt_dlp/downloader/__init__.py]: No. post_live is no longer using the generator protocol

File: yt_dlp/extractor/youtube.py [hunk 3]
```diff
@@ -4123,9 +4160,7 @@ def process_language(container, base_url, lang_code, sub_name, query):
             unified_strdate(get_first(microformats, 'uploadDate'))
             or unified_strdate(search_meta('uploadDate')))
         if not upload_date or (
-            not info.get('is_live')
-            and not info.get('was_live')
-            and info.get('live_status') != 'is_upcoming'
+            live_status in ('not_live', None)
             and 'no-youtube-prefer-utc-upload-date' not in self.get_param('compat_opts', [])
         ):
             upload_date = strftime_or_none(
```
[the remark this answers is not in the document]
Comment (coletdjnz) [reply yt_dlp/extractor/youtube.py:4166]: no post live should not pass.

we don't want utc upload date if:
- video is live
- video was live
- video is upcoming
- video is post live
- compat option is provided

**Pull request merged**